src/hdmf/common/resources.py

```diff
@@ -0,0 +1,116 @@
+import numpy as np
+
+from . import register_class
+from ..container import Table, Container
+
+from ..utils import docval, call_docval_func, popargs
+
+
+def _check_id(table, id):
+    if id >= 0 and len(table.which(id=id)) == 0:
+        return np.uint64(id)
+    else:
+        raise ValueError('id must be a non-negative integer that is not already in the table: %d' % id)
+
+
+@register_class('ResourceReferenceMap')
+class ResourceReferenceMap(Table):
+
+    __defaultname__ = 'resource_map'
+
+    __columns__ = (
+        {'name': 'id', 'type': (int, np.uint64), 'doc': 'The unique identifier in this table.'},
+        {'name': 'key', 'type': str, 'doc': 'The user key that maps to the resource term / registry symbol.'},
+        {'name': 'resource', 'type': str, 'doc': 'The resource/registry that the term/symbol comes from.'},
+        {'name': 'uri', 'type': str, 'doc': 'The unique resource identifier for the resource term / registry symbol.'},
+    )
+
+    @docval(*__columns__)
+    def add_row(self, **kwargs):
+        id = popargs('id', kwargs)
+        kwargs['id'] = _check_id(self, id)
+        return super().add_row(kwargs)
+
+
+@register_class('ResourceReferences')
+class ResourceReferences(Table):
+
+    __defaultname__ = 'references'
+
+    __columns__ = (
+        {'name': 'id', 'type': (int, np.uint64), 'doc': 'The unique identifier in this table.'},
+        {'name': 'object_id', 'type': str, 'doc': 'The UUID for the object that uses this ontology term.'},
+        {'name': 'field', 'type': str,
+         'doc': 'The field from the object (specified by object_id) that uses this ontological term.'},
+        {'name': 'item', 'type': (int, np.uint64),
+         'doc': 'An index into the ResourceReferenceMap that contains the term.'},
+    )
+
+    @docval(*__columns__)
+    def add_row(self, **kwargs):
+        id, item = popargs('id', 'item', kwargs)
+        kwargs['id'] = _check_id(self, id)
+        if item >= 0:
+            kwargs['item'] = np.uint64(item)
+        else:
+            raise ValueError('item must be a non-negative integer: %d' % id)
```

Comment:
```suggestion
            raise ValueError('item must be a non-negative integer: %d' % item)
```

Reply:
It looks like the code to check the item is the same as ``_check_id``. Maybe rename ``_check_id`` to ``_check_index`` and pass in key that needs to be used for the error message.